mkosi/backend.py

```diff
@@ -608,6 +610,13 @@ class MkosiState:
     def root(self) -> Path:
         return self.workspace / "root"
 
+    def init_env(self) -> None:
+        self.environment = dict(self.config.environment)
+        if self.config.image_id is not None:
+            self.environment['IMAGE_ID'] = self.config.image_id
+        if self.config.image_version is not None:
+            self.environment['IMAGE_VERSION'] = self.config.image_version
```

Comment:
No need for this method, we can just pass the environment to the MkosiConfig constructor

Reply:
I believe this is now resolved.